### PR TITLE
Spark framework

### PR DESCRIPTION
**Pull request by tnachen**
Adding the Spark framework

### CODE REVIEW

File: repo/packages/S/spark/0/config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "type": "object",
+  "properties": {
+    "spark/port": {
```
Comment (ssk2) [inline repo/packages/S/spark/0/config.json]: Could we rename this `mesos-port` to avoid potential confusion?

File: repo/meta/index.json [hunk 2]
```diff
@@ -31,6 +31,18 @@
...
         "alpha":"0"
       }
+    },
+    {
+      "currentVersion":"1.3.0",
+      "description":"Spark is a fast and general cluster computing system for Big Data",
+      "name":"spark",
+      "tags":[
+        "mesosphere",
+        "framework"
```
Comment (ssk2) [inline repo/meta/index.json]: Might be nice to have a bigdata/machinelearning tag or something like that. Not sure if that's going against the intended usage of tags though (what do you think, @ConnorDoyle?).

Comment (ConnorDoyle) [inline repo/meta/index.json]: They're just search terms, so anything goes!  Thanks for following precedent by tagging this package as a framework.

File: repo/packages/S/spark/0/config.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      "type": "int",
+      "description": "Port of the Mesos dispatcher.",
+      "default": 7177
```
Comment (ConnorDoyle) [inline repo/packages/S/spark/0/config.json]: Don't forget

```
"minimum": 0,
"maximum": 65535
```

File: repo/packages/S/spark/0/config.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    "spark/webui-port": {
+      "type": "int",
+      "description": "Port of the Web UI for Mesos dispatcher.",
+      "default": 8180
+    }
+  },
+  "required": [
```
Comment (ConnorDoyle) [inline repo/packages/S/spark/0/config.json]: This field can be omitted if the array is empty.

File: repo/packages/S/spark/0/marathon.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "id": "spark",
+  "cpus": 4,
+  "mem": 2048,
+  "instances": 1,
+  "cmd": "./bin/spark-class org.apache.spark.deploy.mesos.MesosClusterDispatcher --port {{spark/port}} --webui-port {{spark/webui-port} --master mesos://master0.test-suite.msphere.co:5050",
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "mesosphere/spark",
```
Comment (ConnorDoyle) [inline repo/packages/S/spark/0/marathon.json]: Please specify a tag here (should correspond to the version in package.json.

File: repo/packages/S/spark/0/config.json (hunk 1 above)
Comment (ConnorDoyle) [inline repo/packages/S/spark/0/config.json]: On the contrary, I like to see a predictable 1:1 correspondence between the command line flag name and the property name.  There's a description, which should serve to disambiguate it from other port-related things.  If anything, maybe the description could be improved a little bit.

File: repo/meta/index.json (hunk 2 above)
Comment (tnachen) [inline repo/meta/index.json]: I'm not even sure what the tags are for, I don't want to start introducing random tags though. Do we have standard on what tags should we be using?

Comment (ConnorDoyle) [inline repo/meta/index.json]: Nothing yet.  If you have ideas, let's add it to the (currently empty) best-practices.md.

File: repo/packages/S/spark/0/config.json (hunk 1 above)
Comment (tnachen) [inline repo/packages/S/spark/0/config.json]: I've just updated the description for now.

Comment (ConnorDoyle) [inline repo/packages/S/spark/0/config.json]: :+1:

**Comment by ConnorDoyle** [comment]
Rebased on `master`.

**Comment by ConnorDoyle** [comment]
:+1:

**Pull request merged**